stage1/init/common/pod.go

```diff
@@ -276,6 +262,116 @@ func findHostPort(pm schema.PodManifest, name types.ACName) uint {
+// of only generating one /etc/{passwd,group} entry
+func generateSysusers(p *stage1commontypes.Pod, ra *schema.RuntimeApp, uid_ int, gid_ int) error {
+	app := ra.App
+	appName := ra.Name
+
+	if err := os.MkdirAll(path.Join(common.Stage1RootfsPath(p.Root), "usr/lib/sysusers.d"), 0755); err != nil {
+		return err
+	}
+
+	gids := append(app.SupplementaryGIDs, gid_)
+
+	// Create the Unix user and group
+	var sysusersConf []string
+
+	for _, g := range gids {
+		groupname := "gen" + strconv.Itoa(g)
+		sysusersConf = append(sysusersConf, fmt.Sprintf("g %s %d\n", groupname, g))
+	}
+
+	username := "gen" + strconv.Itoa(uid_)
+	sysusersConf = append(sysusersConf, fmt.Sprintf("u %s %d \"%s\"\n", username, uid_, username))
+
+	if err := ioutil.WriteFile(path.Join(common.Stage1RootfsPath(p.Root), "usr/lib/sysusers.d", ServiceUnitName(appName)+".conf"),
+		[]byte(strings.Join(sysusersConf, "\n")), 0640); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// lookupPathInsideApp returns the path (relative to the app rootfs) of the
+// given binary. It will look up on "paths" (also relative to the app rootfs)
+// and evaluate possible symlinks to check if the resulting path is actually
+// executable.
+func lookupPathInsideApp(bin string, paths string, appRootfs string) (string, error) {
+	pathsArr := filepath.SplitList(paths)
+	var appPathsArr []string
+	for _, p := range pathsArr {
+		appPathsArr = append(appPathsArr, filepath.Join(appRootfs, p))
+	}
+	for _, path := range appPathsArr {
+		binPath := filepath.Join(path, bin)
+		binAbsPath, err := filepath.Abs(binPath)
+		if err != nil {
+			return "", fmt.Errorf("unable to find absolute path for %s", binPath)
+		}
+		relPath := strings.TrimPrefix(binAbsPath, appRootfs)
+		d, err := os.Lstat(binAbsPath)
+		if err != nil {
+			continue
+		}
+		binRealPath, err := evaluateSymlinksInsideApp(appRootfs, relPath)
+		if err != nil {
+			return "", errwrap.Wrap(fmt.Errorf("could not evaluate path %v", binAbsPath), err)
+		}
+		binRealPath = filepath.Join(appRootfs, binRealPath)
+		d, err = os.Stat(binRealPath)
+		if err != nil {
+			continue
+		}
+		// Check the executable bit, inspired by os.exec.LookPath()
+		if m := d.Mode(); !m.IsDir() && m&0111 != 0 {
+			// The real path is executable, return the path relative to the app
+			return relPath, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find %q in %q", bin, paths)
+}
+
+// appSearchPaths returns a list of paths where we should search for
+// non-absolute exec binaries
+func appSearchPaths(p *stage1commontypes.Pod, appRootfs string, app types.App) []string {
+	appEnv := app.Environment
+
+	if imgPath, ok := appEnv.Get("PATH"); ok {
+		return strings.Split(imgPath, ":")
+	}
+
+	// emulate exec(3) behavior, first check pod's directory and then the list
+	// of directories returned by confstr(_CS_PATH). That's typically
+	// "/bin:/usr/bin" so let's use that.
+	return []string{appRootfs, "/bin", "/usr/bin"}
```

Comment:
*If $PATH is undefined,* emulate exec(3) behavior.

The paths are relative to the stage2 rootfs, so it would be: `{"/", "/bin", "/usr/bin"}`

But `exec(3)` mentions the working directory, so it should be the working directory as defined in the ACI manifest instead of `appRootfs` or "/".

In the comment: `pod's directory`  -->  `the working directory`